### PR TITLE
[flutter_tools] switch to tester with fractional translation and no raster cache

### PR DESCRIPTION
**Pull request by jonahwilliams**
Allow an opt out for the open source framework that requires an opt in for the google3 tooling. Still need to confirm this renders the right things, of course. Either the current golden infra will detect a diff or I'll have to update this PR with a synthetic change to trigger some pixel snapping

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Note: we can't actually land this as is, needs to be opt in to not break g3

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 1]
```diff
@@ -944,6 +954,17 @@ class CachedLocalEngineArtifacts implements LocalEngineArtifacts {
     throw Exception('Unsupported platform $platform.');
   }
 
+  String _flutterTesterFractionalPath(TargetPlatform platform) {
+    if (_platform.isLinux) {
+      return _fileSystem.path.join(engineOutPath, _artifactToFileName(Artifact.flutterTester));
+    } else if (_platform.isMacOS) {
+      return _fileSystem.path.join(engineOutPath, 'flutter_tester_fractional_translation');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/artifacts.dart:961]: Do you know why this is different from `_artifactToFileName(Artifact.flutterTester)`?

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 2]
```diff
@@ -216,6 +216,12 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
               'as a multiplier of the default timeout (e.g. "2x"), '
               'or as the string "none" to disable the timeout entirely.',
         defaultsTo: '30s',
+      )
+      ..addFlag('enable-fractional-translation',
+        help: 'Enables fraction translation of engine composited layers in '
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/test.dart:221]: Should this be hidden with `hide: !verboseHelp`?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/test.dart:221]: Done

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/artifacts.dart]: Don't know, tried to remove it, lets see what happens

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 3]
```diff
@@ -1836,6 +1836,9 @@ abstract class _RenderPhysicalModelBase<T> extends _RenderCustomClip<T> {
     markNeedsPaint();
   }
 
+  @override
+  bool get alwaysNeedsCompositing => true;
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/rendering/proxy_box.dart:1840]: testing

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 4]
```diff
@@ -19,6 +19,8 @@ enum Artifact {
   genSnapshot,
   /// The flutter tester binary.
   flutterTester,
+  /// A flutter tester binary without raster cache or fractional translation.
+  flutterTesterFractionalTranslation,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/artifacts.dart:23]: Is the idea that after a full internal migration, we would go back to only having one flutterTester binary?

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 5]
```diff
@@ -216,6 +216,13 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
...
+              'the software renderer backend for flutter tester. Also disables '
+              'the raster cache.',
+        defaultsTo: true,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/test.dart:224]: Will this break all open source customers' goldens?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/test.dart:224]: if it doesn't break customer-tests, then no :)

**Comment by jonahwilliams** [comment]
still testing to see if this actually works at reducing churn though

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/artifacts.dart:23]: yes, assuming it works 😄

**Comment by jonahwilliams** [comment]
So this improves things a little bit, see https://github.com/flutter/flutter/pull/103783 for a similar code change that caused golden diffs. Unfortunately I didn't turn on SUPPORT_FRACTIONAL_TRANSLATION correctly - we'd need to build an entirely new engine variant to support it. I'm going to look into a runtime flag for that.

**Comment by zanderso** [comment]
> we'd need to build an entirely new engine variant to support it

Let's chat. Maybe we just need to change the way that we build `flutter_tester`. Also, compilation is cheap as long as we can use goma, so if adding a variant is the easiest way to do things, then we shouldn't reject that option.

**Comment by jonahwilliams** [comment]
>  Let's chat. Maybe we just need to change the way that we build flutter_tester. Also, compilation is cheap as long as we can use goma, so if adding a variant is the easiest way to do things, then we shouldn't reject that option.


Definitely - though I think there is some question as to whether or not we should actually be pixel snapping. Of course, removing pixel snapping would be .... breaking. From research, removing pixel snapping regressed iOS32 benchmarks. It might be time to investigate again

**Comment by jonahwilliams** [comment]
See also: https://github.com/flutter/flutter/issues/56722

**Pull request closed without merging**